src/services/shiftRotaService.ts

```diff
@@ -77,4 +78,78 @@ export default class ShiftRotaService {
         if (successCount != shiftData.length) console.log(`${shiftData.length - successCount} error/s occured during the process! Check your data sheet.`)
         return;
     }
-}
\ No newline at end of file
+}
+
+//to convert shift data hours to UTC using UTCTimeConversion function
+let passHoursToUTCTimeConversion = (shiftData:ShiftRota[],offset:number) =>{
+    for (let i = 0; i < shiftData.length; i++) {
+        for (const key in shiftData[i]) {
+          if (key === "date") continue;
+    
+          let time = shiftData[i][key];
+    
+          //for empty times like ""
+          if (time === "") continue;
+          //for times like "9-13;15-16"
+          else if (time.split("").includes(";")) {
+            let subTimes = time.split(";");
+            let convertedSubTimes: string[] = [];
+            subTimes.forEach((time) => {
+              let [startTimeLocal, endTimeLocal]: string[] = time.split("-");
+              let convertedTime = UTCTimeConversion(startTimeLocal, endTimeLocal,offset);
+              convertedSubTimes.push(convertedTime);
+            });
+    
+            shiftData[i][key] = convertedSubTimes.join(";");
+          }
+          //for times like "9-17"
+          else {
+            let [startTimeLocal, endTimeLocal]: string[] = time.split("-");
+            let convertedTime = UTCTimeConversion(startTimeLocal, endTimeLocal,offset);
+    
+            shiftData[i][key] = convertedTime;
+          }
```

Comment:
this can be simplified - if you do str.split(";") on '9-13' you'll get one-element array.
if you do str.split(";") on '9-13;14-17' you'll get two-element array.

I'd just split hours with ';' and iterate through the resulting array adjusting the hours inside (we could get rid of if-else here then and remove a couple of lines potentially)

Reply:
Thanks for this idea - I've implemented this and now splitting with ";" - it has made the function much shorter